gobblin-service/src/test/java/org/apache/gobblin/service/monitoring/FsFlowGraphMonitorTest.java

```diff
@@ -227,6 +234,22 @@ public void testSharedFlowgraphHelper() throws Exception {
   }
 
   @Test (dependsOnMethods = "testSharedFlowgraphHelper")
```

Comment:
how does this test the new logic above? Is it testing the new lock used monitors changes?

Reply:
It's testing that FSFlowGraphMonitor will reload the flowgraph even when there are only changes done to the flow templates, I added some comments to clarify this in the test.

Previously FSFlowGraphMonitor wouldn't listen to the templateCatalog, and would only concern itself on changes made to the flowgraph.